weaver/vault/utils.py

```diff
@@ -0,0 +1,247 @@
+    else:
+        file_id = ""
+    if not file_id:
+        return None
+
+    db = get_db(container)
+    vault = db.get_store(StoreVault)
+    file = vault.get_file(file_id, nothrow=True)
+    href = get_vault_url(file, container)
+    path = get_vault_path(file, container)
+
+    if exists and not os.path.isfile(path):
+        return None
+    return href if url else path
+
+
+def parse_vault_token(header, unique=False):
+    # type: (Optional[str], bool) -> Dict[Optional[str], str]
+    """
+    Parse the authorization header value to retrieve all :term:`Vault` access tokens and optional file UUID.
+
+    .. seealso::
+        - Definition of expected format in :ref:`file_vault_token`.
+        - :class:`sd.XAuthVaultFileHeader`
+
+    :param header: Authorization header to parse.
+    :param unique: Whether only one or multiple tokens must be retrieved.
+    :return: Found mapping of UUID to access token. If unique, UUID can be ``None``.
+    """
+    if not isinstance(header, str):
+        return {}
+    header = header.lower()
+    if unique and "," in header:
+        return {}
+    auth_tokens = header.split(",")
+    if not auth_tokens:
+        return {}
+    if len(auth_tokens) > 1 and unique:  # cannot pick which one applies
+        return {}
+    vault_tokens = {}
+    for auth in auth_tokens:
+        auth = auth.strip()
+        if not unique and ";" not in auth:
+            return {}
+        if unique and ";" not in auth:
+            token = auth
+            param = "="
+        else:
+            token, param = auth.split(";", 1)
+        if param.strip() == "":  # final ';' to ignore
+            param = "="
+        token = token.split("token ")[-1].strip()
+        param = param.split("=")
+        if not len(param) == 2:
+            return {}
+        value = param[1].strip()
+        param = param[0].strip()
+        if param != "id" and not (value or unique):
+            return {}
+        if param and unique and not value:  # explicitly provided parameter although optional is allowed
+            return {}
+        if value and value.startswith("\"") and value.endswith("\""):
+            value = value[1:-1]
+        value_match = REGEX_VAULT_UUID.match(value) if value else None
+        token_match = REGEX_VAULT_TOKEN.match(token) if token else None
+        if not token_match:
+            return {}
+        token_match = token_match[0]
+        if not value_match and not (unique and not value):  # allow omitted 'id' if unique, unless explicitly given
+            return {}
+        value_match = None if (unique and not param) else value_match[0]
+        if vault_tokens.get(value_match) is not None:
+            return {}  # cannot pick duplicates, drop both
+        vault_tokens[value_match] = token_match
+    return vault_tokens
+
+
+def get_vault_auth(request):
+    # type: (Request) -> Tuple[AnyUUID, Optional[str]]
+    """
+    Obtain the requested file if access is granted.
```

Comment:
I don't see any access validation... Looks like it's a copy/paste from the next function.

Reply:
hehe, copy-paste error from below function. Will fix.